core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -995,18 +995,17 @@ protected boolean shouldExecuteReplication(Settings settings) {
 
     static class IndexShardReference implements Releasable {
 
-        final private IndexShard counter;
+        private final Releasable operationLock;
```

Comment:
can we keep the shard reference? I know it's not needed here - but it serves as a proxy to the shard allowing to do more things - see https://github.com/elastic/elasticsearch/pull/15485/files#diff-a8aefbf42f29dc0fcc7c0a144863948eR1104